src/components/apps/launch/ResourceRequirements.js

```diff
@@ -0,0 +1,357 @@
+                name={`requirements.${index}.min_memory_limit`}
+                label={getMessage("minMemory")}
+                component={FormSelectField}
+                renderValue={formatGBValue}
+            >
+                {minMemoryList.map((size, index) => (
+                    <MenuItem key={index} value={size}>
+                        {formatGBListItem(size)}
+                    </MenuItem>
+                ))}
+            </FastField>
+            <FastField
+                id={buildDebugId(baseId, ids.RESOURCE_REQUESTS.MIN_DISK_SPACE)}
+                name={`requirements.${index}.min_disk_space`}
+                label={getMessage("minDiskSpace")}
+                component={FormSelectField}
+                renderValue={formatGBValue}
+            >
+                {minDiskSpaceList.map((size, index) => (
+                    <MenuItem key={index} value={size}>
+                        {formatGBListItem(size)}
+                    </MenuItem>
+                ))}
+            </FastField>
+        </>
+    );
+};
+
+const ResourceRequirementsHeader = ({ headerMessageKey, step }) =>
+    getMessage(headerMessageKey, { values: { step } });
+
+const ResourceRequirementsForm = withI18N(
+    ({
+        baseId,
+        defaultMaxCPUCores,
+        defaultMaxMemory,
+        defaultMaxDiskSpace,
+        limits,
+    }) => {
+        const classes = useStyles();
+
+        return (
+            <ExpansionPanel defaultExpanded={false}>
+                <ExpansionPanelSummary
+                    expandIcon={
+                        <ExpandMore
+                            id={buildDebugId(baseId, ids.BUTTONS.EXPAND)}
+                        />
+                    }
+                >
+                    <Typography variant="subtitle1">
+                        <ResourceRequirementsHeader headerMessageKey="resourceRequirements" />
+                    </Typography>
+                </ExpansionPanelSummary>
+
+                <ExpansionPanelDetails
+                    className={classes.expansionPanelDetails}
+                >
+                    <Typography variant="body1" gutterBottom>
+                        {getMessage("helpMsgResourceRequirements")}
+                    </Typography>
+
+                    {limits.length === 1 ? (
+                        <StepResourceRequirementsForm
+                            baseId={buildDebugId(baseId, limits[0].step_number)}
+                            requirements={limits[0]}
+                            index={0}
+                            defaultMaxCPUCores={defaultMaxCPUCores}
+                            defaultMaxMemory={defaultMaxMemory}
+                            defaultMaxDiskSpace={defaultMaxDiskSpace}
+                        />
+                    ) : (
+                        limits.map((reqs, index) => {
+                            return (
+                                <ExpansionPanel key={reqs.step_number}>
```

Comment:
If there is more than 1 step in the app, then each step gets its own Resource Requirements expansion panel.